modal/image.py

```diff
@@ -553,6 +624,23 @@ def build_dockerfile(version: ImageBuilderVersion) -> DockerfileSpec:
             context_mount=mount,
         )
 
+    def add_local_python_packages(self, *packages: Union[str, Path], copy: bool = False) -> "_Image":
+        """Attaches local Python packages to the container running the image
+
+        Packages are added to the /root directory which is on the PYTHONPATH of any
+        executed Modal functions.
+
+        By default (copy=False) the packages are layered on top of the image when
+        the container starts up and not built as an image layer.
+
+        Set copy=True to force the packages to be added as an image layer instead.
+        This can be slower since it requires a rebuild of the image whenever the
+        required package files change, but it allows you to run additional build
+        steps after this operation.
+        """
+        mount = _Mount.from_local_python_packages(*packages)
```

Comment:
Should we take the opportunity to break with the existing behavior of including data assets and everything else within the directory containing the Python package, at least by default? That's one of our most surprising and problem-causing behaviors IMO.

Reply:
> It's the auto mounts today that exclude everything that's not a .py-file.

I thought auto mounting now included most files? That's how users get themselves into trouble when they automount a project directory that has a lot of data in it.

I agree that you're mostly mirroring the existing behavior of `Mount.from_local_python_packages` but I'm wondering should we take this opportunity to break things about that behavior that have proven problematic.

(I also think we should add the dockerignore conditions to these interfaces, which will help a bit, but my suspicion is that it's easier to debug "this `add_local_python_packages` command is not including my data files" than "my Image build is really slow and seems to be uploading a lot of stuff". But I'm not totally sure.